### ISSUE TITLE
Built app reveals build machine env path on errors? + crashes on Macos High Sierra if built on Big Sur

### ISSUE THREAD

**Issue by ikhmyz** [pyinstaller/pyinstaller#5412]
Traceback (most recent call last):
  File "_pyinstaller_hooks_contrib/hooks/rthooks/pyi_rth_certifi.py", line 13, in <module>
  File "**/Users/BUILD_MACHINE_USER/dev/project_dir/venv/lib/python3.8/site-packages/PyInstaller/loader/pyimod03_importers.py**", line 493, in exec_module
  File "ssl.py", line 98, in <module>
ImportError: dlopen(/Applications/MyApp.app/Contents/MacOS/_ssl.cpython-38-darwin.so, 2): Symbol not found: ____chkstk_darwin
  Referenced from: /Applications/MyApp.app/Contents/MacOS/libcrypto.1.1.dylib (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libSystem.B.dylib
 in /Applications/MyApp.app/Contents/MacOS/libcrypto.1.1.dylib
[2200] Failed to execute script pyi_rth_certifi

Also app was built on Big Sur, but crashes on High Sierra, is there a way to include this lib to make it work on high sierra when built on newer os versions?

**Comment by bwoodsend** [comment]
> Also app was built on Big Sur, but crashes on High Sierra, is there a way to include this lib to make it work on high sierra when built on newer os versions?

This is impossible. Big Sur's system libraries don't really exist and can't be collected.

**Comment by ikhmyz** [comment]
> This is impossible. Big Sur's system libraries don't really exist and can't be collected.

Any ideas how to make app work on High Sierra and Big Sur without making specific builds for each os version? Will app built on older os version work on newer?

**Comment by rokm** [comment]
> This is impossible. Big Sur's system libraries don't really exist and can't be collected.

But we don't collect system libraries, anyway, so libraries being hidden on Big Sur becomes issue only in very special cases.

The problem is indicated in the error message:
>  Referenced from: /Applications/MyApp.app/Contents/MacOS/libcrypto.1.1.dylib (which was built for Mac OS X 10.15)

The libcrypto was built for OSX 10.15 and later, and thus cannot be used on High Sierra (10.13)

**Comment by ikhmyz** [comment]
What about revealing build machine env path in production builds?
It was really surprising to find my username in this error on end user machine and it broke my anonymity

**Comment by rokm** [comment]
And yes, you should generally be building on the oldest version of the OS you want to support.

**Comment by rokm** [comment]
> What about revealing build machine env path in production builds?
It was really surprising to find my username in this error on end user machine and it broke my anonymity

Yes, that's something I've been meaning to look into, because there are some steps in the build process where we are explicitly trying to strip out the full paths from the modules. Apparently it does not (always?) work...

**Comment by rokm** [comment]
Hmm, what version of PyInstaller are you using? If it is not `develop` version, can you try with that? 

Perhaps this was already addressed by #5380...

**Comment by ikhmyz** [comment]
> Hmm, what version of PyInstaller are you using? If it is not `develop` version, can you try with that?
> 
> Perhaps this was already addressed by #5380...

pyinstaller==4.0

This is the version I have adopted my build process, code, libraries versions, patches, hooks to. It always explodes when I'm changing pyinstaller version so it will be super hard to test (but I could try to manually patch with these commits changes)

**Comment by rokm** [comment]
Fair enough :)

I did a quick test, and it seems that since #5380, the paths in errors that originate in pyinstaller modules (such as `exec_module` function in pyimod03_importers.py) are properly stripped. E.g.,

before: 
```
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "/home/rok/tmp/pyi-anonimity/venv/lib64/python3.9/site-packages/PyInstaller/loader/pyimod03_importers.py", line 493, in exec_module
    exec(bytecode, module.__dict__)
```

after:
```
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 359, in exec_module
```

**Comment by rokm** [comment]
I'm closing this because the first part of the issue (libraries built for newer version of macOS not working on older versions of macOS) is not something we can address, and the second part of the issue (PyInstaller's loader modules leaking full paths) has been addressed in PyInstaller 4.2 by #5380.
